gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/binary/types/MapSerializer.java

```diff
@@ -24,20 +24,37 @@
 import org.apache.tinkerpop.gremlin.structure.io.Buffer;
 
 import java.io.IOException;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Set;
 
 public class MapSerializer extends SimpleTypeSerializer<Map> {
+
+    private byte valueFlag = (byte)0x00;
```

Comment:
Although not explicitly stated, I think the serializers are thread-safe because they don't make use of member variables. So don't use a member variable here as it would break that.